src/main/java/frc/utils/auto/LocalADStarAK.java

```diff
@@ -0,0 +1,156 @@
+package frc.utils.auto;
+
+import com.pathplanner.lib.path.GoalEndState;
+import com.pathplanner.lib.path.PathConstraints;
+import com.pathplanner.lib.path.PathPlannerPath;
+import com.pathplanner.lib.path.PathPoint;
+import com.pathplanner.lib.pathfinding.LocalADStar;
+import com.pathplanner.lib.pathfinding.Pathfinder;
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.geometry.Translation2d;
+import org.littletonrobotics.junction.LogTable;
+import org.littletonrobotics.junction.Logger;
+import org.littletonrobotics.junction.inputs.LoggableInputs;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class LocalADStarAK implements Pathfinder {
+
+    private final ADStarIO io = new ADStarIO();
+
+    /**
+     * Get if a new path has been calculated since the last time a path was retrieved
+     *
+     * @return True if a new path is available
+     */
+    @Override
+    public boolean isNewPathAvailable() {
+        if (!Logger.hasReplaySource()) {
+            io.updateIsNewPathAvailable();
+        }
+
+        Logger.processInputs("LocalADStarAK", io);
```

Comment:
maybe save the log path as a constant. i think we have a constants file for these but maybe they are in another CR

Reply:
The constants file is for file directories and not log paths, for log paths we mostly use the class's constants file. I don't want to create a whole constants file for this